keras_nlp/models/roberta/roberta_multi_segment_packer.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+import tensorflow_text as tf_text
+from tensorflow import keras
+
+
+# TODO: This is a temporary, unexported layer until we find a way to make the
+# `MultiSegmentPacker` layer more generic.
+@keras.utils.register_keras_serializable(package="keras_nlp")
+class RobertaMultiSegmentPacker(keras.layers.Layer):
```

Comment:
Thought: we might want to extract some of these methods into utils if we have many packers.

Reply:
Are suggesting utils as opposed to layers? Why?

This is our only "packer layer" we had to pull out, because fairseq does some fairly odd packing here. It's in this dir because only the roberta models need this.

Might be helpful to here more on what you want to accomplish here? (If we want to dig into this more). I had thought we might want to remove this and make our `MultiSegmentPacker` more generic.

Reply:
Sure, we can talk later. Not urgent